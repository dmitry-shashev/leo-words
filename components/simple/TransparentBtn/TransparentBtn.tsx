import { FC } from 'react'

interface Props {
  label: string
  onClick: () => void
  ariaLabel?: string
}

export const TransparentBtn: FC<Props> = ({ label, onClick, ariaLabel }) => {
  return (
    <button
      onClick={onClick}
      aria-label={ariaLabel ?? label}
      className="grow text-lg text-blue-200 border-blue-200 border rounded h-10"
      type="button"
    >
      {label}
    </button>
  )
}
