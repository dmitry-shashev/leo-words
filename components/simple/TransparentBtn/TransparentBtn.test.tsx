import { expect, test, vi } from 'vitest'
import { fireEvent, render } from '@testing-library/react'
import { TransparentBtn } from '@/components/simple/TransparentBtn/TransparentBtn'

test('TransparentBtn', () => {
  const onClick = vi.fn()

  const { asFragment, getByRole } = render(
    <TransparentBtn label="monday" ariaLabel="some btn" onClick={onClick} />
  )

  expect(asFragment()).toMatchSnapshot()

  expect(onClick).toBeCalledTimes(0)
  fireEvent.click(getByRole('button'))
  expect(onClick).toBeCalledTimes(1)
})
