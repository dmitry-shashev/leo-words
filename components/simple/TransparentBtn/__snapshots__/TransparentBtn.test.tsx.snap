// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TransparentBtn 1`] = `
<DocumentFragment>
  <button
    aria-label="some btn"
    class="grow text-lg text-blue-200 border-blue-200 border rounded h-10"
    type="button"
  >
    monday
  </button>
</DocumentFragment>
`;
